xen/xenops_interface.ml

```diff
@@ -580,6 +598,26 @@ module XenopsAPI (R : RPC) = struct
       declare "HOST.upgrade_cpu_features" []
         ( debug_info_p @-> cpu_features_array_p @-> is_hvm_p
         @-> returning cpu_features_array_p err )
+
+    let upgrade_cpu_policy =
+      let is_hvm_p = Param.mk ~name:"is_hvm" Types.bool in
+      declare "Host.upgrade_cpu_policy" ["Given an older cpu feature set, return an updated cpu policy"]
+        (
+            debug_info_p @-> policy_p @-> is_hvm_p @-> returning policy_p err
+        )
+
+    let policy_calc_compatible =
+      declare "HOST.policy_calc_compatible" ["Find out if two host policies are compatible with each other, also returns the intersection"]
+        (
+          debug_info_p @-> policy_p @-> policy_p @-> returning policy_compatibility_p err
+        )
+
+    let policy_is_compatible =
+      declare "HOST.policy_is_compatible" ["Find out if a vm policy is compatible with a host policy, also returns the intersection"]
```

Comment:
Reading the two descriptions is not clear what's the difference between the two. I assumed the only difference is that `is_compatible` returns one of the 3 values of `calc_compatible`

Reply:
`is_compatible` suggests a boolean but the documentation suggests a set is returned. I believe a simple boolean does not require a new definition if the semantics are obvious and so find `policy_boolean_p` dubious.

Reply:
The names of the functions here are based on a draft patch that I had from 'The Xen team' (read, Andrew). We can rename these functions if needed, but I found it easier to keep the names, as I had a patch on hand with the function definitions I could look up.
`is_compatible` will return a boolean for compatibility, but it also _may_ return a message for why the  policies weren't compatible. If they are compatible, it also returns the subset/new policy that makes them so